solr/core/src/java/org/apache/solr/cloud/ShardLeaderElectionContextBase.java

```diff
@@ -185,7 +185,8 @@ void runLeaderProcess(boolean weAreReplacement, int pauseBeforeStartMs)
       DocCollection coll = zkStateReader.getCollection(this.collection);
       if (coll == null || coll.getStateFormat() < 2 || ZkController.sendToOverseer(coll, id)) {
         zkController.getOverseer().offerStateUpdate(Utils.toJSON(m));
-      } else {
+      }
+      if (coll != null && coll.isPerReplicaState()) {
```

Comment:
for prs-collection, will it send message to overseer as well on line 187?

Reply:
It should not because I've updated the sendToOverseer method where I think only in very rare cases, if any would it send to overseer for PRS.

Reply:
I think it will happen for shard split case? would be good to verify that!

Reply:
You are right, split shard it will since it has a parent, but not in normal operation or node up/down cases.